ERCS/erc-7662.md

```diff
@@ -0,0 +1,273 @@
+---
+eip: 7662
+title: AI Agent NFTs
+description: A specification for NFTs that represent AI Agents.
+author: Greg Marlin (@marleymarl)
+discussions-to: https://ethereum-magicians.org/t/erc-7662-ai-agent-nfts/19371
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-03-26
+requires: 721
+---
+
+## Abstract
+
+This proposal introduces a standard for AI agent NFTs. In order for AI Agents to be created and traded as NFTs it doesn't make sense to put the prompts in the token metadata, therefore it requires a standard custom struct. It also needs doesn't make sense to store the prompts directly onchain as they can be quite large, therefore this standard proposes they be stored as decentralized storage URLs. This standard also proposes two options on how this data should be made private to the owner of the NFT, with the favored implementation option being encrypting the data using custom contract parameters for decryption that decrypt only to the owner of the NFT. 
+
+## Motivation
+
+The creation and trading of AI Agent NFTs are a natural fit and offer the potential for an entirely new onchain market. This requires some custom data to be embedded in the NFT through a custom struct and this needs to be standardized so that any marketplace or AI Agent management product, among others, know how to create and parse AI Agent NFTs.  
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+
+All ERC-XXXX compliant contracts MUST implement the standard [ERC-721](./eip-721.md) functionality for minting and transferring NFTs, and MUST additionally implement this standard's Agent struct
```

Comment:
It is unclear how a contract should "implement" a struct. Normally you implement an interface.

Reply:
Will make that update, thanks.